### PR TITLE
Setup event source for my calendar

### PR DESCRIPTION
**Pull request by magnuhar**
### Setup event source for my calendar / my page

This PR introduces an event source to the My Calendar / My Page view. I've done this as simply as possible - so just handled the back-end and set up this, as well as hooked up the My Calendar page to read from the new MyEvents view to get events. Color stuff, date navigation, and all those extra goodies which are useful on a FC setup is not implemented yet. We should discuss what is necessary to add here some time, but I'm not fleshing it out before we have a good and concise plan for this :)

I've tried to be a little clever while writing this one, and have created a mixin that can be re-used on views that serve the purpose of serving events out to fullcalendar instances. From there on we just need to implement the get_queryset method for our specific event source implementation, and the mixin will handle everything from there on. Quite neat.

I also made a new Exception -- UserHasNoPersonException. This exception should be raised when a person dependent functionality is called and the user calling it has no person associated with them. I think this is a good approach.

### CODE REVIEW

File: webook/arrangement/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class UserHasNoPersonException(Exception):
+    pass
```
Comment (eivl) [inline webook/arrangement/exceptions.py:2]: Remember that we need to document this. start by doing it in the Wiki and also make a Kaizen PR to add a docstring to the exception.

File: webook/arrangement/views/calendar_views.py [hunk 2]
```diff
@@ -28,6 +32,56 @@ def __init__(self) -> None:
         self.section = get_section_manifest()
 
 
+
+class EventSourceViewMixin(ListView):
+    """ A mixin for views that serve the express purpose of serving events to FullCalendar calendars """
+
+    # Handle start and end constraining of the queryset in the GET method
+    # Set to False if you want to get all regardless of supplied time constraints, or do your own 
+    # handling of constraints in the get_queryset method.
```
Comment (eivl) [inline webook/arrangement/views/calendar_views.py:41]: No issue to add a docstring on multiple lines, just follow the pep-8 line length for docstrings.

**Review by eivl — APPROVED** [review]

File: webook/arrangement/exceptions.py (hunk 1 above)
Comment (magnuhar) [reply webook/arrangement/exceptions.py:2]: #222

File: webook/arrangement/views/calendar_views.py (hunk 2 above)
Comment (magnuhar) [reply webook/arrangement/views/calendar_views.py:41]: #222

**Pull request merged**